### DISCUSSION TITLE
Deploying Presidio as a nuget package

### DISCUSSION THREAD

**Discussion by palirashmi** [microsoft/presidio#758] [Q&A]
Can I wrap presidio and deploy as a nuget package to be used by a c# application?

**Comment by SharonHart** [comment]
Hi @palirashmi
We have done something similar in the past but haven't released anything official yet.
We did it using [swagger-codegen](https://github.com/swagger-api/swagger-codegen), giving the [Open API spec file](https://github.com/microsoft/presidio/blob/main/docs/api-docs/api-docs.yml) as input and instantly got good results, without any alterations as I recall. I encourage you do to the same until anything official will be released.

**Comment by palirashmi** [comment]
I found something called python.installed where i can actually install pip and then may be presidio-analyzer and anonymizer package.. does that seem like a good option? 
The link to the article is this: I am referring to below image
https://medium.com/scisharp/using-python-libraries-in-net-without-a-python-installation-11124d6190cf

![image](https://user-images.githubusercontent.com/10676371/131175735-8a213e1b-177d-47da-96c7-1bd128b1b552.png)

**Reply by omri374** [reply]
Hi @palirashmi, this sounds like an interesting approach. If you get to test it, please let us know if this worked well. 
The blog post mentions a slowdown of a factor of 4. If this is an issue for you, I would also suggest you try @SharonHart 's approach of running Presidio in docker and calling its API from the C# app.

**Comment by agonzalezm** [comment]
any progress on this? i would like to use this is sdk in c# app as nuget package? 

SpacyDotNet Wrapper is using Python.NET to create a nuget package for spacy net wrapper:

https://spacy.io/universe/project/SpacyDotNet

**Reply by SharonHart** [reply]
@agonzalezm 
Seems like what was done in the Spacy wrapper could work with presidio as well, and should be pretty straight forward implement using the Python.NET package, did you give it a try?
https://pythonnet.github.io/pythonnet/dotnet.html#

**Reply by agonzalezm** [reply]
no, not sure how to start with this. If you can provide official package in this way would be great!

**Reply by agonzalezm** [reply]
@SharonHart Can anyone indicate some instructions on the steps needed to give it a try with python.net?

**Reply by omri374** [reply]
Perhaps using a dockerized version of Presidio and a C# client would be a better option, and would likely have less performance issues from the interop between python and C#. Is that an option?

**Reply by agonzalezm** [reply]
not really because we need to run this on embedded agent on a windows system that cant run docker linux images.

**Reply by omri374** [reply]
@agonzalezm providing an official C# package is not currently on our roadmap. We'd be happy to review community contributions on this.
